base/linalg/cholesky.jl

```diff
@@ -491,6 +491,21 @@ function A_ldiv_B!(C::CholeskyPivoted, B::StridedMatrix)
     end
 end
 
+\(::Cholesky, ::RowVector) = throw(DimensionMismatch("Cannot left-divide matrix by transposed vector"))
+\(::CholeskyPivoted, ::RowVector) = throw(DimensionMismatch("Cannot left-divide matrix by transposed vector"))
+\(::Cholesky{<:Number}, ::RowVector{<:Number}) = throw(DimensionMismatch("Cannot left-divide matrix by transposed vector"))
```

Comment:
@andyferris My question was mainly about the need for both a method for `Cholesky` and one for `Cholesky{<:Number}`. Anyway, given the `*` decision, I think it makes sense just to remove these definitions and just throw `DimensionMismatch`.